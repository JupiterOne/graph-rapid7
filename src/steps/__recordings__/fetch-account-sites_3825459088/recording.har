{
  "log": {
    "_recordingName": "fetch-account-sites",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "944d602e73c3e23c6ab13c3eecd1cd4e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "localhost:3780"
            }
          ],
          "headersSize": 278,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://localhost:3780/api/3/administration/info"
        },
        "response": {
          "bodySize": 1117,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 1117,
            "text": "{\n  \"cpu\" : {\n    \"clockSpeed\" : 2591,\n    \"count\" : 8\n  },\n  \"disk\" : {\n    \"free\" : {\n      \"bytes\" : 34408022016,\n      \"formatted\" : \"32 GB\"\n    },\n    \"installation\" : {\n      \"backups\" : {\n        \"bytes\" : 0,\n        \"formatted\" : \"0 bytes\"\n      },\n      \"database\" : {\n        \"bytes\" : 2587800330,\n        \"formatted\" : \"2.4 GB\"\n      },\n      \"directory\" : \"/opt/rapid7/nexpose\",\n      \"reports\" : {\n        \"bytes\" : 0,\n        \"formatted\" : \"0 bytes\"\n      },\n      \"scans\" : {\n        \"bytes\" : 13924,\n        \"formatted\" : \"13.6 KB\"\n      },\n      \"total\" : {\n        \"bytes\" : 6304460333,\n        \"formatted\" : \"5.9 GB\"\n      }\n    },\n    \"total\" : {\n      \"bytes\" : 94164131840,\n      \"formatted\" : \"87.7 GB\"\n    }\n  },\n  \"distinguishedName\" : \"CN=Rapid7 Security Console, O=Rapid7\",\n  \"fqdn\" : \"8a15d4ef4d06\",\n  \"host\" : \"8a15d4ef4d06\",\n  \"ip\" : \"172.17.0.2\",\n  \"jvm\" : {\n    \"name\" : \"OpenJDK 64-Bit Server VM\",\n    \"startTime\" : \"2023-08-09T16:57:21.585Z\",\n    \"uptime\" : \"PT48M56.603S\",\n    \"vendor\" : \"Azul Systems, Inc.\",\n    \"version\" : \"25.372-b07\"\n  },\n  \"links\" : [ {\n    \"href\" : \"https://localhost:3780/api/3/administration/info\",\n    \"rel\" : \"self\"\n  } ],\n  \"memory\" : {\n    \"free\" : {\n      \"bytes\" : 257859584,\n      \"formatted\" : \"245.9 MB\"\n    },\n    \"total\" : {\n      \"bytes\" : 16788279296,\n      \"formatted\" : \"15.6 GB\"\n    }\n  },\n  \"operatingSystem\" : \"Ubuntu Linux 18.04\",\n  \"serial\" : \"014DF13E1BAEA3C0B414AE7B199FAE30814F4B49\",\n  \"superuser\" : true,\n  \"user\" : \"root\",\n  \"version\" : {\n    \"build\" : \"70\",\n    \"changeset\" : \"8853f1e3bc8590c03f04ffdbc44fead2bd62a0de\",\n    \"platform\" : \"Linux64\",\n    \"semantic\" : \"6.6.210\",\n    \"update\" : {\n      \"content\" : \"1896038050\",\n      \"contentPartial\" : \"Unknown\",\n      \"id\" : {\n        \"productId\" : \"299067162755562\",\n        \"versionId\" : \"490\"\n      },\n      \"product\" : \"356135944\"\n    }\n  }\n}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-ua-compatible",
              "value": "IE=edge,chrome=1"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-store, must-revalidate"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "date",
              "value": "Wed, 09 Aug 2023 17:46:17 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Security Console"
            }
          ],
          "headersSize": 377,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-08-09T17:46:17.048Z",
        "time": 1071,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1071
        }
      },
      {
        "_id": "2ece270a1243a9d9f9b60c1312e8ea8b",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "localhost:3780"
            }
          ],
          "headersSize": 280,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "0"
            },
            {
              "name": "size",
              "value": "500"
            }
          ],
          "url": "https://localhost:3780/api/3/sites?page=0&size=500"
        },
        "response": {
          "bodySize": 873,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 873,
            "text": "{\n  \"resources\" : [ {\n    \"assets\" : 1,\n    \"id\" : 1,\n    \"importance\" : \"normal\",\n    \"lastScanTime\" : \"2023-08-09T17:34:52.846Z\",\n    \"links\" : [ {\n      \"href\" : \"https://localhost:3780/api/3/sites/1\",\n      \"rel\" : \"self\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/alerts\",\n      \"rel\" : \"Alerts\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/scan_engine\",\n      \"rel\" : \"Scan Engine\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/scan_schedules\",\n      \"rel\" : \"Schedules\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/organization\",\n      \"rel\" : \"Organization\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/tags\",\n      \"rel\" : \"Tags\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/users\",\n      \"rel\" : \"Users\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/scan_template\",\n      \"rel\" : \"Template\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/site_credentials\",\n      \"rel\" : \"Site Credentials\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/shared_credentials\",\n      \"rel\" : \"Assigned Shared Credentials\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/web_authentication/html_forms\",\n      \"rel\" : \"Web HTML Forms Authentication\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/web_authentication/http_headers\",\n      \"rel\" : \"Web HTTP Headers Authentication\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/assets\",\n      \"rel\" : \"Assets\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/included_targets\",\n      \"rel\" : \"Included Targets\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/excluded_targets\",\n      \"rel\" : \"Excluded Targets\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/included_asset_groups\",\n      \"rel\" : \"Included Asset Groups\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/sites/1/excluded_asset_groups\",\n      \"rel\" : \"Excluded Asset Groups\"\n    } ],\n    \"name\" : \"vulnsite\",\n    \"riskScore\" : 18994.0,\n    \"scanEngine\" : 3,\n    \"scanTemplate\" : \"full-audit-without-web-spider\",\n    \"type\" : \"static\",\n    \"vulnerabilities\" : {\n      \"critical\" : 10,\n      \"moderate\" : 0,\n      \"severe\" : 42,\n      \"total\" : 52\n    }\n  } ],\n  \"page\" : {\n    \"number\" : 0,\n    \"size\" : 500,\n    \"totalResources\" : 1,\n    \"totalPages\" : 1\n  },\n  \"links\" : [ {\n    \"href\" : \"https://localhost:3780/api/3/sites?page=0&size=500&sort=id,asc\",\n    \"rel\" : \"self\"\n  } ]\n}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-ua-compatible",
              "value": "IE=edge,chrome=1"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-store, must-revalidate"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "date",
              "value": "Wed, 09 Aug 2023 17:46:18 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Security Console"
            }
          ],
          "headersSize": 377,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-08-09T17:46:18.162Z",
        "time": 154,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 154
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
