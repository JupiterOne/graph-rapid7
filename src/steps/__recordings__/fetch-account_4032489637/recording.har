{
  "log": {
    "_recordingName": "fetch-account",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "944d602e73c3e23c6ab13c3eecd1cd4e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "localhost:3780"
            }
          ],
          "headersSize": 278,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://localhost:3780/api/3/administration/info"
        },
        "response": {
          "bodySize": 1117,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 1117,
            "text": "{\n  \"cpu\" : {\n    \"clockSpeed\" : 2591,\n    \"count\" : 8\n  },\n  \"disk\" : {\n    \"free\" : {\n      \"bytes\" : 34407989248,\n      \"formatted\" : \"32 GB\"\n    },\n    \"installation\" : {\n      \"backups\" : {\n        \"bytes\" : 0,\n        \"formatted\" : \"0 bytes\"\n      },\n      \"database\" : {\n        \"bytes\" : 2587800330,\n        \"formatted\" : \"2.4 GB\"\n      },\n      \"directory\" : \"/opt/rapid7/nexpose\",\n      \"reports\" : {\n        \"bytes\" : 0,\n        \"formatted\" : \"0 bytes\"\n      },\n      \"scans\" : {\n        \"bytes\" : 13924,\n        \"formatted\" : \"13.6 KB\"\n      },\n      \"total\" : {\n        \"bytes\" : 6304494271,\n        \"formatted\" : \"5.9 GB\"\n      }\n    },\n    \"total\" : {\n      \"bytes\" : 94164131840,\n      \"formatted\" : \"87.7 GB\"\n    }\n  },\n  \"distinguishedName\" : \"CN=Rapid7 Security Console, O=Rapid7\",\n  \"fqdn\" : \"8a15d4ef4d06\",\n  \"host\" : \"8a15d4ef4d06\",\n  \"ip\" : \"172.17.0.2\",\n  \"jvm\" : {\n    \"name\" : \"OpenJDK 64-Bit Server VM\",\n    \"startTime\" : \"2023-08-09T16:57:21.585Z\",\n    \"uptime\" : \"PT48M57.702S\",\n    \"vendor\" : \"Azul Systems, Inc.\",\n    \"version\" : \"25.372-b07\"\n  },\n  \"links\" : [ {\n    \"href\" : \"https://localhost:3780/api/3/administration/info\",\n    \"rel\" : \"self\"\n  } ],\n  \"memory\" : {\n    \"free\" : {\n      \"bytes\" : 256196608,\n      \"formatted\" : \"244.3 MB\"\n    },\n    \"total\" : {\n      \"bytes\" : 16788279296,\n      \"formatted\" : \"15.6 GB\"\n    }\n  },\n  \"operatingSystem\" : \"Ubuntu Linux 18.04\",\n  \"serial\" : \"014DF13E1BAEA3C0B414AE7B199FAE30814F4B49\",\n  \"superuser\" : true,\n  \"user\" : \"root\",\n  \"version\" : {\n    \"build\" : \"70\",\n    \"changeset\" : \"8853f1e3bc8590c03f04ffdbc44fead2bd62a0de\",\n    \"platform\" : \"Linux64\",\n    \"semantic\" : \"6.6.210\",\n    \"update\" : {\n      \"content\" : \"1896038050\",\n      \"contentPartial\" : \"Unknown\",\n      \"id\" : {\n        \"productId\" : \"299067162755562\",\n        \"versionId\" : \"490\"\n      },\n      \"product\" : \"356135944\"\n    }\n  }\n}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-ua-compatible",
              "value": "IE=edge,chrome=1"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-store, must-revalidate"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "date",
              "value": "Wed, 09 Aug 2023 17:46:18 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Security Console"
            }
          ],
          "headersSize": 377,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-08-09T17:46:18.615Z",
        "time": 592,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 592
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
