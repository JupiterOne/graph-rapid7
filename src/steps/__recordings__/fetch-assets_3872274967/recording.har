{
  "log": {
    "_recordingName": "fetch-assets",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "f59ffd89f2ee8282e0fa618158d3456d",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "localhost:3780"
            }
          ],
          "headersSize": 281,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page",
              "value": "0"
            },
            {
              "name": "size",
              "value": "500"
            }
          ],
          "url": "https://localhost:3780/api/3/assets?page=0&size=500"
        },
        "response": {
          "bodySize": 1129,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 1129,
            "text": "{\n  \"resources\" : [ {\n    \"addresses\" : [ {\n      \"ip\" : \"172.18.0.3\",\n      \"mac\" : \"02:42:AC:12:00:03\"\n    } ],\n    \"assessedForPolicies\" : false,\n    \"assessedForVulnerabilities\" : true,\n    \"history\" : [ {\n      \"date\" : \"2023-08-09T17:33:45.998Z\",\n      \"scanId\" : 1,\n      \"type\" : \"SCAN\",\n      \"version\" : 1\n    } ],\n    \"hostName\" : \"da7a8ca02894.vulnerable-net\",\n    \"hostNames\" : [ {\n      \"name\" : \"da7a8ca02894.vulnerable-net\",\n      \"source\" : \"dns\"\n    } ],\n    \"id\" : 1,\n    \"ip\" : \"172.18.0.3\",\n    \"links\" : [ {\n      \"href\" : \"https://localhost:3780/api/3/assets/1\",\n      \"rel\" : \"self\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/software\",\n      \"rel\" : \"Software\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/files\",\n      \"rel\" : \"Files\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/users\",\n      \"rel\" : \"Users\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/user_groups\",\n      \"rel\" : \"User Groups\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/databases\",\n      \"rel\" : \"Databases\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/services\",\n      \"rel\" : \"Services\"\n    }, {\n      \"href\" : \"https://localhost:3780/api/3/assets/1/tags\",\n      \"rel\" : \"Tags\"\n    } ],\n    \"mac\" : \"02:42:AC:12:00:03\",\n    \"os\" : \"Debian Linux\",\n    \"osCertainty\" : \"0.8\",\n    \"osFingerprint\" : {\n      \"description\" : \"Debian Linux\",\n      \"family\" : \"Linux\",\n      \"id\" : 1,\n      \"product\" : \"Linux\",\n      \"systemName\" : \"Debian Linux\",\n      \"vendor\" : \"Debian\"\n    },\n    \"rawRiskScore\" : 18994.46875,\n    \"riskScore\" : 18994.46875,\n    \"services\" : [ {\n      \"configurations\" : [ {\n        \"name\" : \"http.banner\",\n        \"value\" : \"Apache/2.4.25 (Debian)\"\n      }, {\n        \"name\" : \"http.banner.server\",\n        \"value\" : \"Apache/2.4.25 (Debian)\"\n      } ],\n      \"family\" : \"Apache\",\n      \"links\" : [ {\n        \"href\" : \"https://localhost:3780/api/3/assets/1/services/tcp/80\",\n        \"rel\" : \"self\"\n      }, {\n        \"href\" : \"https://localhost:3780/api/3/assets/1/services/tcp/80/configurations\",\n        \"rel\" : \"Configurations\"\n      }, {\n        \"href\" : \"https://localhost:3780/api/3/assets/1/services/tcp/80/databases\",\n        \"rel\" : \"Databases\"\n      }, {\n        \"href\" : \"https://localhost:3780/api/3/assets/1/services/tcp/80/users\",\n        \"rel\" : \"Users\"\n      }, {\n        \"href\" : \"https://localhost:3780/api/3/assets/1/services/tcp/80/user_groups\",\n        \"rel\" : \"User Groups\"\n      }, {\n        \"href\" : \"https://localhost:3780/api/3/assets/1/services/tcp/80/web_applications\",\n        \"rel\" : \"Web Applications\"\n      } ],\n      \"name\" : \"HTTP\",\n      \"port\" : 80,\n      \"product\" : \"HTTPD\",\n      \"protocol\" : \"tcp\",\n      \"vendor\" : \"Apache\",\n      \"version\" : \"2.4.25\"\n    } ],\n    \"vulnerabilities\" : {\n      \"critical\" : 10,\n      \"exploits\" : 4,\n      \"malwareKits\" : 0,\n      \"moderate\" : 0,\n      \"severe\" : 42,\n      \"total\" : 52\n    }\n  } ],\n  \"page\" : {\n    \"number\" : 0,\n    \"size\" : 500,\n    \"totalResources\" : 1,\n    \"totalPages\" : 1\n  },\n  \"links\" : [ {\n    \"href\" : \"https://localhost:3780/api/3/assets?page=0&size=500&sort=id,asc\",\n    \"rel\" : \"self\"\n  } ]\n}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-ua-compatible",
              "value": "IE=edge,chrome=1"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-store, must-revalidate"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "date",
              "value": "Wed, 09 Aug 2023 17:46:17 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "Security Console"
            }
          ],
          "headersSize": 377,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2023-08-09T17:46:17.106Z",
        "time": 327,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 327
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
