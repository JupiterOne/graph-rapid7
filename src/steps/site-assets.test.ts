import { steps } from '../constants';
import { buildStepTestConfig } from '../../test/config';
import {
  Recording,
  executeStepWithDependencies,
} from '@jupiterone/integration-sdk-testing';
import { setupRapid7Recording } from '../../test/helpers/recording';

let recording: Recording;

afterEach(async () => {
  if (recording) {
    await recording.stop();
  }
});

describe('fetch-site-assets', () => {
  test('success', async () => {
    recording = setupRapid7Recording({
      name: 'fetch-site-assets',
      directory: __dirname,
      options: {
        recordFailedRequests: false,
      },
    });

    const stepConfig = buildStepTestConfig(steps.FETCH_SITE_ASSETS);
    const stepResults = await executeStepWithDependencies(stepConfig);
    expect(stepResults).toMatchSnapshot();
  });
});
