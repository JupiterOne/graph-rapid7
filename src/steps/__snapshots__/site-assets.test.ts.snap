// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch-site-assets success 1`] = `
Object {
  "collectedData": Object {},
  "collectedEntities": Array [],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "insightvm_site:1",
      "_key": "insightvm_site:1|has|insightvm_asset:1",
      "_toEntityKey": "insightvm_asset:1",
      "_type": "insightvm_site_has_asset",
      "displayName": "HAS",
    },
  ],
  "encounteredEntityKeys": Set {
    "insightvm_account:root",
    "insightvm_site:1",
    "insightvm_asset:1",
  },
  "encounteredTypes": Array [
    "insightvm_account",
    "insightvm_site",
    "insightvm_asset",
    "insightvm_site_has_asset",
  ],
}
`;
